tfjs-backend-webgpu/src/index.ts

```diff
@@ -34,8 +35,21 @@ tf.registerBackend('webgpu', async () => {
   };
 
   const adapter = await navigator.gpu.requestAdapter(gpuDescriptor);
-  const device = await adapter.requestDevice({});
-  return new WebGPUBackend(device, glslang);
+  let isSupportQueryAPI =
+    adapter.extensions.includes('timestamp-query' as GPUExtensionName);
+
+  if (isSupportQueryAPI) {
+    const ENV = env();
+    ENV.registerFlag('WEBGPU_QUERY_API_ENABLED', () => true);
+    const deviceDescriptor: GPUDeviceDescriptor = {
+      extensions: ['timestamp-query' as GPUExtensionName]
+    };
+    const device = await adapter.requestDevice(deviceDescriptor);
+    return new WebGPUBackend(device, glslang);
```

Comment:
You can directly pass isSupportQueryAPI to WebGPUBackend instead of using a env flag.

Reply:
Done